src/librustc/hir/mod.rs

```diff
@@ -46,9 +47,17 @@ impl<'tcx> TyCtxt<'tcx> {
     pub fn hir(self) -> Hir<'tcx> {
         Hir { tcx: self, map: &self.hir_map }
     }
+
+    pub fn hir_id_parent_module(self, id: HirId) -> DefId {
```

Comment:
Maybe `parent_hir_module` would be more appropriate.

Reply:
Maybe renaming the query to `parent_module_from_def_id` and letting this take `parent_module` is a better option?